tests/textdata_preprocesser_test.py

```diff
@@ -0,0 +1,146 @@
+        impression_sentences = ["No acute cardiopulmonary abnormality."]
+        impression_text = ''.join(impression_sentences)
+        pack = self.iu_xray_pl.process(document)
+        for idx, impression in enumerate(pack.get(Impression)):
+            self.assertEqual(impression.text, impression_text)
+
+
+class TestNonAlphaTokenRemover(unittest.TestCase):
+    r"""
+    Test iu xray processor for non alpha token removal in content
+    """
+    def setUp(self):
+        self.iu_xray_pl = Pipeline[MultiPack]()
+        self.iu_xray_pl.set_reader(StringReader())
+        self.iu_xray_pl.add(NLTKSentenceSegmenter())
+        self.iu_xray_pl.add(NLTKWordTokenizer())
+        self.iu_xray_pl.add(MultiPackBoxer())
+        self.iu_xray_pl.add(NonAlphaTokenRemover())
+        self.iu_xray_pl.initialize()
+
+    def test_cleaner(self):
+        sentences = ["Free intraperitoneal air.  ___ was successfully "
+                     "paged to discuss this finding on ___ at 3:10 p.m. "
+                     "at the time of discovery."]
+        document = ' '.join(sentences)
+        nonalpha_token_rm_sentences = ["Free intraperitoneal air. was "
+                                       "successfully paged to discuss "
+                                       "this finding on at pm at the "
+                                       "time of discovery."]
+
+        rm_text = ' '.join(nonalpha_token_rm_sentences)
+        pack = self.iu_xray_pl.process(document)
+        for idx, sentences in enumerate(pack.get(Sentence)):
+            self.assertEqual(sentences.text, rm_text)
+
+
+class TestBuildPipeline(unittest.TestCase):
+    r"""
+    Test iu xray processor for non alpha token removal in content
+    """
+    def setUp(self):
+        self.result_dir = 'tests/test_xml/generated_xml'
+        self.iu_xray_pl = build_pipeline(self.result_dir)
+
+        self.ground_truth_findings = ''.join([
+            'the lungs and pleural spaces show no acute abnormality. '
+            'lungs are mildly hyperexpanded. heart size and pulmonary '
+            'vascularity within normal limits.'
+        ])
+        self.ground_truth_impression = 'no acute pulmonary abnormality.'
+
+    def test_pipeline(self):
+        for _ in self.iu_xray_pl.process_dataset('tests/test_xml'):
+            pass
+
+        for i, filename in enumerate(os.listdir(self.result_dir)):
+            self.assertIn(filename, ['CXR333_IM-1594-1001.json', 'CXR333_IM-1594-2001.json'])
+            with open(osp.join(self.result_dir, filename), 'r') as f:
+                items = list(DataPack.deserialize(f.read()))
+                # print(DataPack.deserialize(f.read()).get('Findings'))
+                key = filename.replace('.json', '')
+                self.assertEqual(items[0].text, self.ground_truth_findings)
+                self.assertEqual(items[1].text, self.ground_truth_impression)
+                self.assertEqual(items[2].img_study_path, key)
+                f.close()
```

Comment:
Hmm, I believe the context manager will close the `f`, so we don't need to call `close`, actually I am not sure if this will be a potential problem

Reply:
see, I will remove the ``f.close()``